server/api/runtime_handlers/mpijob/v1.py

```diff
@@ -270,6 +270,15 @@ def _resolve_crd_object_status_info(
             )
         return in_terminal_state, completion_time, desired_run_state
 
+    def _resolve_container_error_status(self, crd_object: dict) -> tuple[str, str]:
```

Comment:
It seems like this function is duplicated from the one you added in `server/api/runtime_handlers/base.py`. It could be moved to a helper module or something similar?

Reply:
they may look similar but they are not the same. the runtime resource structure is different